src/Symfony/Component/DependencyInjection/Tests/Compiler/IntegrationTest.php

```diff
@@ -699,6 +708,88 @@ static function (Definition $definition, CustomAutoconfiguration $attribute) {
         ], $collector->collectedTags);
     }
 
+    /**
+     * @requires PHP 8
+     */
+    public function testTagsViaAttributeOnPropertyMethodAndParameter()
+    {
+        $container = new ContainerBuilder();
+        $container->registerAttributeForAutoconfiguration(
+            CustomMethodAttribute::class,
+            static function (ChildDefinition $definition, CustomMethodAttribute $attribute, ReflectionMethod $reflector) {
+                $tagAttributes = get_object_vars($attribute);
+                $tagAttributes['method'] = $reflector->getName();
+
+                $definition->addTag('app.custom_tag', $tagAttributes);
+            }
+        );
+        $container->registerAttributeForAutoconfiguration(
+            CustomPropertyAttribute::class,
+            static function (ChildDefinition $definition, CustomPropertyAttribute $attribute, ReflectionProperty $reflector) {
+                $tagAttributes = get_object_vars($attribute);
+                $tagAttributes['property'] = $reflector->getName();
+
+                $definition->addTag('app.custom_tag', $tagAttributes);
+            }
+        );
+        $container->registerAttributeForAutoconfiguration(
+            CustomParameterAttribute::class,
+            static function (ChildDefinition $definition, CustomParameterAttribute $attribute, ReflectionParameter $reflector) {
+                $tagAttributes = get_object_vars($attribute);
+                $tagAttributes['parameter'] = $reflector->getName();
+
+                $definition->addTag('app.custom_tag', $tagAttributes);
+            }
+        );
+        $container->registerAttributeForAutoconfiguration(
+            CustomAnyAttribute::class,
+            eval(<<<'PHP'
+            return static function (\Symfony\Component\DependencyInjection\ChildDefinition $definition, \Symfony\Component\DependencyInjection\Tests\Fixtures\Attribute\CustomAnyAttribute $attribute, \ReflectionClass | \ReflectionMethod | \ReflectionProperty | \ReflectionParameter $reflector) {
```

Comment:
no spaces around `|`

Reply:
Weird, because fabpot requires me to do this (in another PR)? curl https://fabbot.io/patch/symfony/symfony/42051/f38cc2e8c03ec6215d28c97e7ec534b2d9be6010/cs.diff | patch -p0

Reply:
yes,fabbot has an issue right now, we're waiting for a release new of php-cs-fixer